ocs_ci/utility/workloads/jenkins.py

```diff
@@ -0,0 +1,79 @@
+"""
+This module implements all the functionalities required for setting up and
+running "Jenkins" like workloads on the pods.
+
+This module implements few functions
+setup(): for setting up git utility on the pod and any necessary
+    environmental params.
+run(): for running 'git clone' on pod to simulate a working environment for
+    a developer
+
+Note: The above mentioned functions will be invoked from Workload.setup()
+and Workload.run() methods along with user provided parameters.
+"""
+import logging
+
+from ocs_ci.ocs import exceptions
+
+
+log = logging.getLogger(__name__)
+
+DISTROS = {"Debian": "apt-get", "RHEL": "yum"}
+
+
+def find_distro(io_pod):
+    """
+    Find whats the os distro on pod
+
+    Args:
+        io_pod (Pod): app pod object
+
+    Returns:
+        distro (str): representing 'Debian' or 'RHEL' as of now
+    """
+    for distro, pkg_mgr in DISTROS.items():
+        try:
+            io_pod.exec_cmd_on_pod(f"which {pkg_mgr}", out_yaml_format=False)
+        except exceptions.CommandFailed:
+            log.debug(f"Distro is not {distro}")
+        else:
+            return distro
+
+
+def setup(**kwargs):
+    """
+    setup git workload
+
+    Args:
+        **kwargs (dict): git setup configuration.
+            The only argument present in kwargs is 'pod' on which we
+            want to setup
+
+    Returns:
+        bool: True if setup succeeds else False
+    """
+    io_pod = kwargs['pod']
+    # For first cut doing simple fio install
+    distro = find_distro(io_pod)
+    pkg_mgr = DISTROS[distro]
+
+    if distro == 'Debian':
+        cmd = f'{pkg_mgr} update'
+        io_pod.exec_cmd_on_pod(cmd, out_yaml_format=False)
+
+    cmd = f"{pkg_mgr} -y install git"
+    return io_pod.exec_cmd_on_pod(cmd, out_yaml_format=False)
+
+
+def run(**kwargs):
+    """
+    Run git clone
+
+    Result:
+        result of command
+    """
+    io_pod = kwargs.pop('pod')
+    git_clone_cmd = "git clone https://github.com/rook/rook.git"
```

Comment:
we can make this configurable so that we can use different git endpoints.

Reply:
done